src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -535,22 +549,31 @@ public IntervalList getLociToGenotype(final Collection<Fingerprint> fingerprints
             final HaplotypeBlock haplotypeBlock = this.haplotypes.getHaplotype(info.getSequenceName(), info.getPosition());
             final Snp snp = this.haplotypes.getSnp(info.getSequenceName(), info.getPosition());
 
-            for (final SamLocusIterator.RecordAndOffset rec : info.getRecordAndOffsets()) {
+            // randomly select locusMaxReads elements from the list
+            final List<SamLocusIterator.RecordAndOffset> recordAndOffsetList = locusMaxReads == 0 ?
+                    info.getRecordAndOffsets() :
+                    MathUtil.randomSublist(info.getRecordAndOffsets(), locusMaxReads, random);
+
+            for (final SamLocusIterator.RecordAndOffset rec : recordAndOffsetList) {
                 final SAMReadGroupRecord rg = rec.getRecord().getReadGroup();
                 final FingerprintIdDetails details;
-                if (rg == null || !fingerprintIdDetailsMap.containsKey(rg)) {
+
+                // if there's a missing rg tag, add the unknown FP id to the map
+                if (rg == null && !fingerprintIdDetailsMap.containsKey(null)) {
+
                     final FingerprintIdDetails unknownFPDetails = createUnknownFP(samFile, rec.getRecord());
+
                     fingerprintIdDetailsMap.put(null, unknownFPDetails);
 
                     final Fingerprint fp = new Fingerprint(unknownFPDetails.sample, samFile, unknownFPDetails.platformUnit);
                     fingerprintsByReadGroup.put(unknownFPDetails, fp);
 
                     for (final HaplotypeBlock h : this.haplotypes.getHaplotypes()) {
-                        fp.add(new HaplotypeProbabilitiesFromSequence(h));
+                        fp.add(blockToProbMapper.apply(h));
                     }
                 }
 
-                if (fingerprintIdDetailsMap.containsKey(rg)) {
+                if (rg == null || fingerprintIdDetailsMap.containsKey(rg)) {
```

Comment:
If `rg == null` then `fingerprintIdDetailsMap.containsKey(rg)` is always true because the `if` block above adds `null` to the map. So it seems like the `rg == null` test is unnecessary.